File: src/main/java/io/github/toberocat/core/listeners/PlayerDeathListener.java
package io.github.toberocat.core.listeners;

import io.github.toberocat.core.factions.Faction;
import io.github.toberocat.core.factions.FactionUtility;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

public class PlayerDeathListener implements Listener {

    @EventHandler
    public void death(PlayerDeathEvent event) {
        Faction faction = FactionUtility.getPlayerFaction(event.getEntity());
        if (faction == null) return;

        faction.getPowerManager().death();
    }
}

File: improved-factions/src/main/kotlin/io/github/toberocat/improvedfactions/modules/power/listener/PlayerDeathListener.kt
package io.github.toberocat.improvedfactions.modules.power.listener

import io.github.toberocat.improvedfactions.database.storage.StorageManager
import io.github.toberocat.improvedfactions.modules.base.BaseModule
import io.github.toberocat.improvedfactions.modules.power.impl.FactionPowerRaidModuleHandleImpl
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

class PlayerDeathListener(private val powerRaidModuleHandleImpl: FactionPowerRaidModuleHandleImpl) : Listener {

    @EventHandler
    private fun onDeath(event: PlayerDeathEvent) {
        if (event.entity.world.name !in BaseModule.config.allowedWorlds)
            return

        val factionId = StorageManager.cache.user(event.entity.uniqueId)?.factionId ?: return
        val faction = StorageManager.cache.faction(factionId) ?: return
        powerRaidModuleHandleImpl.playerDie(faction)
    }
}
